extensions/resource-deployment/src/services/resourceTypeService.ts

```diff
@@ -0,0 +1,175 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import { ResourceType, ResourceTypeOption, DeploymentProvider } from '../interfaces';
+import { IToolsService } from './toolsService';
+import * as fs from 'fs';
+import * as path from 'path';
+
+export interface IResourceTypeService {
+	getResourceTypes(jsonFilePath: string): ResourceType[];
+	validateResourceTypes(resourceTypes: ResourceType[]): string[];
+}
+
+export const PackageJsonPath = path.join(__dirname, '../../package.json');
+
+export class ResourceTypeService implements IResourceTypeService {
+	constructor(private toolsService: IToolsService) { }
+
+	/**
+	 * Get the supported resource types
+	 * @param jsonFilePath the path of the json file that contains the resourceTypes node.
+	 * @param filterByPlatform indicates whether to return the resource types supported on current platform.
+	 */
+	getResourceTypes(jsonFilePath: string, filterByPlatform: boolean = true): ResourceType[] {
+		if (!fs.existsSync(jsonFilePath)) {
```

Comment:
I'd suggest caching this so we don't have to read the file in each time this is called